lib/fs/filesystem_test.go

```diff
@@ -171,3 +173,56 @@ func TestIsParent(t *testing.T) {
 		}
 	}
 }
+
+func TestRepro9677(t *testing.T) {
```

Comment:
Pet peeve, explain what the test is supposed to test :)

Reply:
Added something to the name that could give a hint, and wrote a comment trying to explain it.

Doing so I noticed another thing: We create an FS key for cached FS that includes options. So a FS with and without mtimefs should produce distinct keys, preventing everything we saw?! Clearly I am either misunderstanding what that key mechanism does or it's broken. I'll have a look at that too.

Reply:
Yeah, mtime FS "swallows" it's own option, thus it's not included in the cache key. So instead of only caching the cache, I could also fix the cache key, in one of two ways (or both): Include the types of underlying/wrapped filesystems in the key. And/or add the mtime option to the `Filesystem.Options` result. I'll probably do both regardless, seems like the right thing to do.